### DISCUSSION TITLE
List of incompatibilities from the compatibility matrix

### DISCUSSION THREAD

**Discussion by yeachan153** [fabric8io/kubernetes-client#4790] [Q&A]
Hi all,

After reading the [compatibility matrix](https://github.com/fabric8io/kubernetes-client#compatibility-matrix), we were wondering if there is a page which lists a description of what exactly those incompatibilities are?

**Comment by rohanKanojia** [comment]
You're right. We should document this.

I think the incompatibilities refer to the new resources / API groups that get added in each new version of Kubernetes. KubernetesClient would work okay with standard resources on any specific Kubernetes cluster. However, it won't guarantee that it will support all the Kubernetes resources (available in `kubectl api-resources`) provided by that Kubernetes Cluster.

**Reply by yeachan153** [reply]
OK, thanks for clarifying :-)

**Reply by manusa** [reply]
Note the "won't guarantee". Version 6 (and latter versions of 5) should be compatible with any resource available/accessible for your user's credentials in the cluster, at the very least to be handled as a Generic (untyped/raw) resource.
